torch/lib/THCS/generic/THCSTensor.c

```diff
@@ -459,8 +464,57 @@ void THCTensor_(sparseMask)(THCState *state, THCSTensor *r_, THCTensor *t, THCST
   THCudaLongTensor_free(state, indicesBuffer);
   THLongStorage_free(viewSize);
   THCTensor_(free)(state, t_view);
-  THCIndexTensor_(free)(state, maskIndices);
-  THCTensor_(free)(state, maskValues);
+}
+
+void THCTensor_(sparseCopy)(THCState *state, THCTensor *r_, THCTensor *dense, THCSTensor *sparse) {
+  THCAssertSameGPU(THCSTensor_(checkGPU)(state, 1, 3, sparse, r_, dense));
+
+  THCTensor *r = r_;
+  if (r != dense) {
+    THCTensor_(retain)(state, r);
+    THCTensor_(resizeAs)(state, r, dense);
+    THCTensor_(copy)(state, r, dense);
+  } else {
+    r = THCTensor_(newContiguous)(state, r_);
+  }
+
+  const ptrdiff_t nnz = THCSTensor_(nnz)(state, sparse);
+  if (nnz == 0) {
+    THCTensor_(freeCopyTo)(state, r, r_);
```

Comment:
Maybe just move the `nnz`check before you start assigning to `r` and remove this `freeCopyTo`? You'll save a bunch of unnecessary ops this way